contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -92,6 +93,18 @@
 
   private static final String LONG_COLUMN_NAME = "long_data";
 
+  private static final String DATA_SIZE_COLUMN_NAME = "data_size";
+
+  private static final String ELEMENT_COUNT_NAME = "element_count";
+
+  private static final String DATASET_DATA_TYPE_NAME = "dataset_data_type";
+
+  private static final String DIMENSIONS_FIELD_NAME = "dimensions";
+
+  private static final int PREVIEW_ROW_LIMIT = 20;
+
+  private static final int MAX_DATASET_SIZE = 16777216; // 16MB
```

Comment:
Nit: this can be better expressed as `16 * 1024 * 1024`. Or, since you are mimicking the max buffer size, set your constant to `ValueVector.MAX_BUFFER_SIZE`.

Reply:
Fixed